url8867e6a4e2_cyberpython_JAwesomeChart_tgz-pJ8-com_gmigdos_jawesomechart_renderers_twodimensional_horizontal_ColumnChartRendererJ8/src/com/gmigdos/jawesomechart/core/DataSeriesList.java

```diff
@@ -0,0 +1,238 @@
+        if(size()==0){return 0.0;}
+        List<Double> minVals = new ArrayList<Double>();
+        for (DataSeries dataSeries : this) {
+            minVals.add(dataSeries.getMinValue());
+        }
+        return Collections.min(minVals);
+    }
+    
+    /**
+     * Returns the minimum of the first elements of each series 
+     * (first column if each series is considered to be a row) in this list.
+     * This method can be useful for {@link ChartRenderer}s that use only a single 
+     * value from each series.
+     * 
+     * @return the minimum of the first elements of each series in this list.
+     */
+    public Double getMinValueOnFirstColumn(){
+        if(size()==0){return 0.0;}
+        List<Double> vals = new ArrayList<Double>();
+        for (DataSeries dataSeries : this) {
+            DataSeriesDataProvider dp = dataSeries.get(0);
+            if(dp!=null){
+                vals.add(dp.getDataValue());
+            }
+        }
+        return Collections.min(vals);
+    }
+    
+    /**
+     * Returns the sum of the positive first elements of each series 
+     * (first column if each series is considered to be a row) in this list.
+     * This method can be useful for {@link ChartRenderer}s that use only a single 
+     * value from each series.
+     * 
+     * @return the sum of the positive first elements of each series in this list.
+     */
+    public double getSumOfPositivesOnFirstColumn(){
+        double sum = 0;
+        double value;
+        for (DataSeries dataSeries : this) {
+            DataSeriesDataProvider dp = dataSeries.get(0);
+            if(dp!=null){
+                value = dp.getDataValue();
+                if(value>0){
+                    sum+=value;
+                }
+            }
+        }
+        return sum;
+    }
+    
+    /**
+     * Returns the number of the positive first elements of each series 
+     * (first column if each series is considered to be a row) in this list.
+     * This method can be useful for {@link ChartRenderer}s that use only a single 
+     * value from each series.
+     * 
+     * @return the number of the positive first elements of each series in this list.
+     */
+    public int getPositiveCountOnFirstColumn(){
+        int count = 0;
+        for (DataSeries dataSeries : this) {
+            DataSeriesDataProvider dp = dataSeries.get(0);
+            if(dp!=null){
+                if(dp.getDataValue()>=0){
+                    count++;
+                }
+            }
+        }
+        return count;
+    }
+    
+    /**
+     * Returns the length of the longest series in this list.
+     * @return the length of the longest series in this list
+     */
+    public int getMaxDataSeriesLength(){
+        int max = 0;
+        int size;
+        for (DataSeries dataSeries : this) {
+            size = dataSeries.size();
+            if(dataSeries.size()>max){
+                max = size;
+            }
+        }
+        return max;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 192:200, 208:216.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.